lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1489,17 +1779,44 @@ static bool diagnoseNoReturn(ADContext &context, SILFunction *original,
   return true;
 }
 
-/// If the original function in the differentiation task has more than one basic
-/// blocks, emit a "control flow unsupported" error at appropriate source
-/// locations. Returns true if error is emitted.
+/// If the original function contains unsupported control flow, emit a "control
+/// flow unsupported" error at appropriate source locations. Returns true if
+/// error is emitted.
+///
+/// Update as control flow support is added. Currently, loops and branching
+/// terminators other than `br` and `cond_br` are not supported.
 static bool diagnoseUnsupportedControlFlow(ADContext &context,
```

Comment:
Will this function be called multiple times? `SILLoopInfo` constructor does the loop analysis on the function, and we don't want to compute it more than once.

Reply:
We discussed in person. It turns out there's `SILLoopAnalysis`, which returns `SILLoopInfo` for functions. I'll use that instead.